runtime/src/lib.rs

```diff
@@ -1339,6 +1339,25 @@ impl_runtime_apis! {
             Asset::PoolShare(SerdeWrapper(pool_id))
         }
     }
+
+    #[cfg(feature = "try-runtime")]
+    impl frame_try_runtime::TryRuntime<Block> for Runtime {
+        fn on_runtime_upgrade() -> (frame_support::weights::Weight, frame_support::weights::Weight) {
+            log::info!("try-runtime::on_runtime_upgrade.");
+            // NOTE: intentional unwrap: we don't want to propagate the error backwards, and want to
+            // have a backtrace here. If any of the pre/post migration checks fail, we shall stop
+            // right here and right now.
+            let weight = Executive::try_runtime_upgrade().map_err(|err|{
+                log::info!("try-runtime::on_runtime_upgrade failed with: {:?}", err);
+                err
+            }).unwrap();
```

Comment:
```suggestion
            let weight = Executive::try_runtime_upgrade().unwrap();
```

This one confused me quite a bit. While `Executive::try_runtime_upgrade()` does return a `Result<Weight, 'static &str>`, it _cannot_ return an `Err`. See https://paritytech.github.io/substrate/master/src/frame_executive/lib.rs.html#265-272. If pre/post checks fail, it panics. Otherwise, it returns `Ok(weight)`. So we never get to see the message `"try-runtime::on_runtime_upgrade failed with: {:?}"`.

If you want to keep the message (insurance against future changes of `try_runtime_upgrade`), you can also do this, which should be funcionally equivalent to your suggestion:

```suggestion
            let weight = Executive::try_runtime_upgrade().expect(
                "try-runtime::on_runtime_upgrade failed with"
            );
```

Reply:
That's true. My second suggestion was merely meant as a functionally equivalent alternative to what you already have (in case `try_runtime_upgrade` ever actually returns an error).